packages/docregistry/src/registry.ts

```diff
@@ -988,6 +988,11 @@ export namespace DocumentRegistry {
      * It should emit the [widgetCreated] signal with the new widget.
      */
     createNew(context: IContext<U>): T;
+
+    /**
+     * Clone a new widget given a context
+     */
+    clone(context: IContext<U>): T;
```

Comment:
I had to do some typecasting for docManager.cloneWidget to achieve this- I'd really like some feedback on whether that's the right approach.

Reply:
The changes look great! I think we ought to just make `cloneWidget` only take in `IDocumentWidget`s instead of typecasting them. If you can make that change, I'm good with merging this. Thanks again!